packages/app/src/systems/Chains/eth/hooks/useBlock.tsx

```diff
@@ -0,0 +1,31 @@
+import { useQuery } from '@tanstack/react-query';
+
+import { useEthAccountConnection } from './useEthAccountConnection';
+
+function calculateBlockAge(timestamp?: number) {
+  if (!timestamp) {
+    return 'N/A';
+  }
+  const currentDate = new Date();
+  const blockDate = new Date(timestamp * 1000);
+  const diffInTime = currentDate.getTime() - blockDate.getTime();
+  const diffInDays = Math.round(diffInTime / (1000 * 3600 * 24));
+  return `${diffInDays} days ago`;
+}
```

Comment:
can we use `date-fns` to calculate age instead of doing it natively ?

```
import { formatDistanceToNow } from 'date-fns';

const timestamp = 1621712123000; // Replace with your actual timestamp

const distance = formatDistanceToNow(timestamp, { addSuffix: true });
```